File: Data Structures & Algorithms/coin-change/submission-2.java
class Solution {
    public int coinChange(int[] coins, int amount) 
    {
        int n = coins.length;
        int dp[][] = new int[amount+1][n+1];
        for(int i=1;i<=amount;i++){
            for(int j = 0;j<=n;j++){
                dp[i][j] = Integer.MAX_VALUE/2;
            }
        }
        for(int i = 1;i<=amount;i++){
            for(int j = 1;j<=n;j++){
                int not_added = dp[i][j-1];
                int added = Integer.MAX_VALUE/2;
                if(i>=coins[j-1]){
                    added = 1 + dp[i-coins[j-1]][j];
                }
                
                dp[i][j] = Math.min(added,not_added);
            }
        }
        
        int ans = dp[amount][n];
        return ans >= Integer.MAX_VALUE / 2 ? -1 : ans;
        
    }
}

File: Data Structures & Algorithms/min-cost-climbing-stairs/submission-2.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        n = len(cost)
        # ans = [0]*(n)
        # ans[0] = cost[0]
        # ans[1] = cost[1]
        # if n == 2:
        #     return min(cost[0],cost[1])
        for i in range(2,n):
            cost[i] = cost[i] + min(cost[i-1],cost[i-2])
        return min(cost[n-1],cost[n-2])
        